File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        newS=''
        revs=''

        if not s:
            return true

        for i in s:
            if i.isalnum():
                newS+=i.lower()
            else:
                continue
        revS= newS[::-1]#reverse a string

        return newS==revS

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.empty()){
            return true;
        }
        string newStr;
        for(int i=0; i<s.size(); i++){
            if(isalnum(s[i])){
                newStr += tolower(s[i]);
            }
        }

        cout<<newStr;
        int l=0, r=newStr.size()-1;

        while(l<=r){
            if(newStr[l]!=newStr[r]){
                return false;
            }
            l++;
            r--;
        }

        return true;
    }
};